### PR TITLE
Ddf sql

### PR DESCRIPTION
**Pull request by PangZhi**
support for sql. Adding more support for sql grammar now.

### CODE REVIEW

**Comment by khangich** [comment]
ok to test

File: core/src/main/java/io/ddf/content/SqlTypedResult.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package io.ddf.content;
+
+
+import java.sql.SQLType;
```
Comment (lebinh) [inline core/src/main/java/io/ddf/content/SqlTypedResult.java]: this interface is added in Java 8 but our build and runtime Java version is only 7.x atm

Comment (phvu) [inline core/src/main/java/io/ddf/content/SqlTypedResult.java]: @PangZhi can you fix this please, this code doesn't compile with java 7.

Comment (phvu) [inline core/src/main/java/io/ddf/content/SqlTypedResult.java]: We might actually consider to move to java 8. Java 8 has many nice features.

File: core/src/main/java/io/ddf/TableVisitor.java [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+import net.sf.jsqlparser.statement.select.*;
+import net.sf.jsqlparser.statement.show.ShowColumns;
+import net.sf.jsqlparser.statement.show.ShowTables;
+import net.sf.jsqlparser.statement.truncate.Truncate;
+import net.sf.jsqlparser.statement.update.Update;
+
+import java.util.Iterator;
+
+/**
+ * Created by jing on 6/29/15.
```
Comment (Huandao0812) [inline core/src/main/java/io/ddf/TableVisitor.java]: remove these @PangZhi

**Comment by Huandao0812** [comment]
pls add test suite for this @PangZhi

File: core/src/main/java/io/ddf/DDFManager.java [hunk 3]
```diff
@@ -334,12 +338,42 @@ public DDF sql2ddf(String command, Schema schema, DataFormat dataFormat) throws
 
   @Override
   public DDF sql2ddf(String command, Schema schema, String dataSource, DataFormat dataFormat) throws DDFException {
```
Comment (Huandao0812) [inline core/src/main/java/io/ddf/DDFManager.java]: what is dataFormat for?

File: jdbc/examples/jdbc.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import io.ddf.datasource.{DataSourceURI, JDBCDataSourceDescriptor}
+import io.ddf.datasource.JDBCDataSourceDescriptor.JDBCDataSourceCredentials
+import io.ddf.jdbc.JDBCDDFManager
+import io.ddf.misc.Config.ConfigConstant
+
+System.setProperty(ConfigConstant.JDBC_DRIVER_PROPERTY.toString, "cdata.jdbc.salesforce.SalesforceDriver")
+
+val sourceDescriptor = new JDBCDataSourceDescriptor(
+  new DataSourceURI("jdbc:salesforce:User=bhan@adatao.com;Password=KualaLumpur123!@#;SecurityToken=OgBwt0V2NU3fltKAse4sJdmga;"),
+  new JDBCDataSourceCredentials("bhan@adatao.com", "KualaLumpur123!@#"),
```
Comment (nhanitvn) [inline jdbc/examples/jdbc.scala]: We should not have real credentials in such an open-source project.

Comment (namma) [inline jdbc/examples/jdbc.scala]: wow, this is bad..

**Pull request closed without merging**